sympy/solvers/inequalities.py

```diff
@@ -982,3 +985,487 @@ def reduce_inequalities(inequalities, symbols=[]):
+    column vector B represent a set of constraints. For example,
+        1*x + 0*y <= 2
+        0*x + 0*y <= 4
+    can be represented as:
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> A = Matrix([[1, 0],
+    ...             [0, 1]])
+    >>> B = Matrix([[2],
+    ...             [4]])
+
+    In this interpretation, row vector C is an objective function to be
+    maximized. For example,
+        3x + y
+    can be represented as:
+
+    >>> C = Matrix([[3, 1]])
+
+    Parameters
+    ==========
+
+    Under the maximization interpretation, m is the number of constraints and n
+    is the number of variables.
+
+    A : Matrix with shape (m, n)
+        Must contain only Rational and float coefficients.
+
+    B : Matrix with shape (m, 1)
+        Must contain only Rational and float coefficients.
+
+    C: Matrix with shape (1, n)
+        Must contain only Rational and float coefficients.
+
+    Returns
+    =======
+
+    Returns `(optimum, argmax, and argmax_dual)`:
+
+    optimum : float or Rational
+        maximum value of objective function under the constraints
+
+    argmax : list of floats and/or Rationals
+        x
+
+    argmax_dual : list of floats and/or Rationals
+        y
+
+    Examples
+    ========
+
+    Suppose we want to maximize
+        3x + y
+    subject to
+        x + y <= 2
+
+    >>> from sympy.matrices.dense import Matrix
+    >>> from sympy.solvers.inequalities import linprog_from_matrices
+    >>> A = Matrix([[1, 1]])
+    >>> B = Matrix([[2]])
+    >>> C = Matrix([[3, 1]])
+    >>> optimum, argmax, argmax_dual = linprog_from_matrices(A, B, C)
+    >>> optimum
+    6
+    >>> argmax
+    [2, 0]
+    >>> argmax_dual
+    [3]
+
+    Suppose we want to maximize
+        3x + y
+    subject to
+        x <= 2
+        y <= 4
+
+    >>> A = Matrix([[1 ,0], [0, 1]])
+    >>> B = Matrix([[2], [4]])
+    >>> C = Matrix([[3, 1]])
+    >>> optimum, argmax, argmax_dual = linprog_from_matrices(A, B, C)
+    >>> optimum
+    10
+    >>> argmax
+    [2, 4]
+    >>> argmax_dual
+    [3, 1]
+
+    See Also
+    ========
+
+    linprog_maximize_from_equations
+    find_feasible
+    """
+    from sympy.matrices.dense import Matrix
+
+    A, B, C = [Matrix(i) for i in (A, B, C)]
+
+    m, n = A.shape
+    if B.shape != (m, 1) or C.shape != (1, n):
+        raise ValueError(f"The shape of matrix B ({B.shape}) or C ({C.shape})" \
+                         f"does not match the shape of matrix A ({A.shape}).")
+    return _simplex(A, B, C)
+
+
+def _linear_programming_to_matrix(constraints, objective, variables):
+    """
+    Converts a list of constraints and an objective function into the standard
+    form for linear programming.
+
+    Examples
+    ========
+
+    >>> from sympy.solvers.inequalities import _linear_programming_to_matrix
+    >>> from sympy.abc import x, y, z
+    >>> from sympy import Eq
+    >>> r1 = y + 2*z <= 3
+    >>> r2 = -x - 3*z <= -2
+    >>> r3 = 2*x + y + 7*z <= 5
```

Comment:
```suggestion
    >>> r3 = y - 5 >= 2*x + 7*z
```

Reply:
I like this suggestion but I think it should be `5 - y`, not `y - 5`.

1. 2x + y + 7z <= 5
2. y - 5 <= -2x - 7z (move the variables to different sides of the equation)
3. 5 - y >= 2x + 7z (multiply by -1)